### PR TITLE
CVE-2023-44487 boefje

### PR DESCRIPTION
**Pull request by ammar92**
### Changes
Boefje and normalizer for CVE-2023-44487

### Proof
image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [ ] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by dekkers — COMMENTED** [review]
This code will create a CVE finding for every HTTP/2 server. I don't think this is very useful. As far as I understand it every compliant HTTP/2 support RST_STREAM because that is part of the HTTP/2 specification. `SAFE` is only returned when the server doesn't support HTTP/2, and `LIKELY` and `POSSIBLE` are returned when an error is encountered.

The readme of the original exploit has also been updated to reflect this, see also https://github.com/bcdannyboy/CVE-2023-44487/issues/7#issuecomment-1760265385 and https://github.com/bcdannyboy/CVE-2023-44487/issues/1#issuecomment-1760268937

Basically the code checks whether the server is susceptible to attack, e.g. if it implements RST_STREAM as specified in the HTTP/2 spec. It doesn't check how it handles a larger number of RST_STREAM requests and I don't think we can check this without actually doing the DoS attack.

I don't think we should create a CVE finding for every HTTP/2 server and keep that finding forever even if the server is patched or was never vulnerable.

**Comment by underdarknl** [comment]
@dekkers correct, that is what it does for now, and I'd rather *have* the warning, than not have the warning. Once we get better signatures available, or once we get a more precise test (without hammering the server until the boefje trips a firewall rule) this is an OK path to follow for a zero day.  Obviously, we could also wait till everyone has patched their servers, and there's a surgical test available, but that would be more or less a meaningless addition to OpenKAT at that point.
Tldr, this boefje will get improved capabilities once the community at large developers better tests / signatures.

**Comment by dekkers** [comment]
It is currently already meaningless in my opinion because it would just add a CVE finding to every web server that is not so ancient that it doesn't implement HTTP/2. And it doesn't add a warning, it adds a finding that the server is vulnerable to CVE-2023-44487. And we already know that this wrong for widely used webservers such as nginx and haproxy which don't seem to be impacted in the default configuration and for webservers that are patched. 

In my opinion we should aim for higher quality than this and we should make sure that we don't generate high number of false positives.

**Comment by underdarknl** [comment]
Ok, then we'll not warn people.
